mmdet/datasets/pipelines/transforms.py

```diff
@@ -494,6 +495,291 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class MosaicCrop(object):
+
+    def __init__(self):
+        self.bbox2label = {
+            'gt_bboxes': 'gt_labels',
+            'gt_bboxes_ignore': 'gt_labels_ignore'
+        }
+
+    def __call__(self, results, w, h, pleft, ptop, swidth, sheight):
+        for key in results.get('img_fields', ['img']):
+            img = results[key]
+            aug_img = self.image_data_augmentation(img, pleft, ptop, swidth,
+                                                   sheight)
+            results[key] = aug_img
+        results['img_shape'] = aug_img.shape
+
+        for key in results.get('bbox_fields', []):
+            bboxes = results[key]
+            label_key = self.bbox2label.get(key)
+            if label_key in results:
+                labels = results[label_key]
+            else:
+                labels = None
+
+            aug_bboxes, aug_labels = self.bboxes_data_augmentation(
+                bboxes, labels, w, h, pleft, ptop, swidth, sheight)
+
+            results[key] = aug_bboxes
+            if label_key in results:
+                results[label_key] = aug_labels
+
+        return results
+
+    @staticmethod
+    def bboxes_data_augmentation(bboxes, labels, w, h, pleft, ptop, swidth,
```

Comment:
Avoid such general names. Name a function so that people can grasp the purpose of it immediately.